### PR TITLE
feat: add test case use netty with system proxy

### PR DESCRIPTION
**Pull request by ChenTanyi**

### CODE REVIEW

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java [hunk 1]
```diff
@@ -224,6 +234,44 @@ public void afterTest() throws IOException {
...
+                        }
+                    }
+                }
+                String host = null;
+                int port = 0;
+                if (System.getProperty("https.proxyHost") != null && System.getProperty("https.proxyPort") != null) {
+                    host = System.getProperty("https.proxyHost");
+                    port = Integer.parseInt(System.getProperty("https.proxyPort"));
+                } else if (System.getProperty("http.proxyHost") != null && System.getProperty("http.proxyPort") != null) {
+                    host = System.getProperty("https.proxyHost");
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java]: http.proxyHost?

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java [hunk 2]
```diff
@@ -224,6 +234,44 @@ public void afterTest() throws IOException {
...
+                    port = Integer.parseInt(System.getProperty("http.proxyPort"));
+                }
+                if (host != null) {
+                    clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress(host, port)));
```
Comment (weidongxu-microsoft) [inline azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java]: Why http/https both map to HTTP, not HTTPS?

Comment (ChenTanyi) [reply azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java]: The proxy type has http only.

File: azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java (hunk 1 above)
Comment (ChenTanyi) [reply azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java]: done

**Comment by ChenTanyi** [comment]
It did work for no system proxy.

**Pull request merged**